src/extension/params-extractor/src/main/java/org/geoserver/params/extractor/EchoParametersDao.java

```diff
@@ -83,52 +95,44 @@ public static void saveOrUpdateEchoParameter(EchoParameter echoParameter) {
 
     public static void saveOrUpdateEchoParameter(
             EchoParameter echoParameter, Resource input, Resource output) {
-        try (InputStream inputStream = input.in();
```

Comment:
combining input and output will also leave an empty output file if reading fails. Better to separate this. It's also easier to read.